### PR TITLE
D3D: Only try to apply exclusive mode when the renderer is in focus.

### PR DESCRIPTION
**Pull request by Armada651**
This should prevent users from ending up in an exclusive mode loop. The previous code would try to immediately re-apply exclusive mode after losing it.

This PR is a rewrite of the exclusive fullscreen handling in the Direct3D backend.

### CODE REVIEW

**Comment by Armada651** [comment]
@degasus Could you review this PR? It's a stability patch that only requires you to review the logic and whether you agree with adding another host function.

**Comment by degasus** [comment]
I'm always against Host functions, but I think this one here is necessary.

File: Source/Core/DolphinWX/X11Utils.cpp [hunk 1]
```diff
@@ -38,7 +38,12 @@ void ToggleFullscreen(Display *dpy, Window win)
 	// Send the event
 	if (!XSendEvent(dpy, DefaultRootWindow(dpy), False,
 	                SubstructureRedirectMask | SubstructureNotifyMask, &event))
+	{
 		ERROR_LOG(VIDEO, "Failed to switch fullscreen/windowed mode.");
```
Comment (degasus) [inline Source/Core/DolphinWX/X11Utils.cpp]: I guess this error here is a generic x11 issue. This doesn't say if fullscreen is now enabled or not. To determ if it's enabled, you would have to spend a round-trip to the x server.

Comment (Armada651) [inline Source/Core/DolphinWX/X11Utils.cpp]: We don't use the function for X11 anyway, so I think right now just some basic error handling is enough.

**Pull request merged**